src/Razor/src/Microsoft.VisualStudio.LanguageServer.ContainedLanguage/PublicAPI.Unshipped.txt

```diff
@@ -1,6 +1,17 @@
+Microsoft.VisualStudio.LanguageServer.ContainedLanguage.MessageInterception.InterceptionMiddleLayer2
+Microsoft.VisualStudio.LanguageServer.ContainedLanguage.MessageInterception.InterceptionMiddleLayer2.CanHandle(string! methodName) -> bool
+Microsoft.VisualStudio.LanguageServer.ContainedLanguage.MessageInterception.InterceptionMiddleLayer2.HandleNotificationAsync<TMessage>(string! methodName, TMessage message, System.Func<TMessage, System.Threading.Tasks.Task!>! sendNotification) -> System.Threading.Tasks.Task!
+Microsoft.VisualStudio.LanguageServer.ContainedLanguage.MessageInterception.InterceptionMiddleLayer2.HandleRequestAsync<TRequest, TResponse>(string! methodName, TRequest methodParam, System.Func<TRequest, System.Threading.Tasks.Task<TResponse?>!>! sendRequest) -> System.Threading.Tasks.Task<TResponse?>!
+Microsoft.VisualStudio.LanguageServer.ContainedLanguage.MessageInterception.InterceptionMiddleLayer2.InterceptionMiddleLayer2(Microsoft.VisualStudio.LanguageServer.ContainedLanguage.MessageInterception.InterceptorManager! interceptorManager, string! contentType) -> void
+Microsoft.VisualStudio.LanguageServer.ContainedLanguage.MessageInterception.InterceptionResult.ChangedToken.get -> object?
+Microsoft.VisualStudio.LanguageServer.ContainedLanguage.MessageInterception.InterceptionResult.InterceptionResult(object? newToken, bool changedDocumentUri) -> void
+Microsoft.VisualStudio.LanguageServer.ContainedLanguage.MessageInterception.InterceptorManager.ProcessInterceptorsAsync(string! methodName, Newtonsoft.Json.Linq.JToken! message, string! contentType, System.Threading.CancellationToken cancellationToken) -> System.Threading.Tasks.Task<Newtonsoft.Json.Linq.JToken?>!
 virtual Microsoft.VisualStudio.LanguageServer.ContainedLanguage.LSPDocument.UpdateVirtualDocument<TVirtualDocument>(TVirtualDocument! virtualDocument, System.Collections.Generic.IReadOnlyList<Microsoft.VisualStudio.Text.ITextChange!>! changes, int hostDocumentVersion, object? state) -> Microsoft.VisualStudio.LanguageServer.ContainedLanguage.LSPDocumentSnapshot!
 virtual Microsoft.VisualStudio.LanguageServer.ContainedLanguage.LSPDocumentManager.RefreshVirtualDocuments() -> void
+virtual Microsoft.VisualStudio.LanguageServer.ContainedLanguage.MessageInterception.MessageInterceptor.ApplyChangesAsync(Newtonsoft.Json.Linq.JToken! message, string! containedLanguageName, System.Threading.CancellationToken cancellationToken) -> System.Threading.Tasks.Task<Microsoft.VisualStudio.LanguageServer.ContainedLanguage.MessageInterception.InterceptionResult>!
+virtual Microsoft.VisualStudio.LanguageServer.ContainedLanguage.MessageInterception.MessageInterceptor.ApplyChangesAsync<T>(T message, string! containedLanguageName, System.Threading.CancellationToken cancellationToken) -> System.Threading.Tasks.Task<Microsoft.VisualStudio.LanguageServer.ContainedLanguage.MessageInterception.InterceptionResult>!
```

Comment:
I'm surprised to see new public surface area -- especially with `NewtonSoft.Json` in there. Is this needed?

Reply:
The new methods her that take a JToken aren't actually new, they're moving from abstract to virtual. Trying to prevent a source breaking change, and hoping it isn't a binary breaking change. Was hoping to use this PR/build to verify against Web Tools, to answer your question above, but seems like CI doesn't like it. Will have to set Web Tools up locally I guess.

Reply:
Update: Web Tools builds against this with only obsolete errors, so I think the source breaking bit is at least okay. Will have to wait for a real package to test binary break, our CI doesn't have access to the devdiv feed for the bits we're referencing here.

Reply:
Thanks for the update!